phpBB/phpbb/controller/helper.php

```diff
@@ -364,7 +364,7 @@ protected function set_cron_task()
 			if ($task)
 			{
 				$url = $task->get_url();
-				$this->template->assign_var('RUN_CRON_TASK', '<img src="' . $url . '" width="1" height="1" alt="cron" />');
+				$this->template->assign_var('RUN_CRON_TASK', '<img src="' . $url . '" width="1" height="1" alt="" style="position: fixed; top: -10000px" />');
```

Comment:
If you are bound to modify this you shold know that the inline styling  is to be avoided nowadays, not to mention that the IMG tag does not not need a closing ` />` but `>`

Reply:
> you shold know that the inline styling is to be avoided nowadays

In general, yes; I left it inline because it seems all external CSS files are only loaded for the user's currently selected style, e.g. ProSilver, whereas this should be set for all styles. Any suggestions for where it should go?

> IMG tag does not not need a closing `/>` but `>`

I left it as-is for consistency with elsewhere in core, but I'll change if we're moving away from that style 👍

Reply:
> I left it as-is for consistency with elsewhere in core, but I'll change if we're moving away from that style 👍

Right or wrong, when changing a line of code that belongs to the core, as it is old code, it is a good habit to improve it.

Reply:
I notice all the HTML examples in https://area51.phpbb.com/docs/33x/coding-guidelines.html still use trailing slashes for void elements (`<br />` etc), and also that https://www.phpbb.com/community/ still proudly claims "Valid XHTML 1.0 Strict" in its footer, which couldn't possibly be true if there were void elements without the closing slash. Are there some more comprehensive and up-to-date style guidelines somewhere that clarify this stuff?

Reply:
It is valid: https://www.w3.org/TR/xhtml1/#h-4.6

Where do you get that it isn't?

Reply:
Nope, there are not. All of that is very old code and guidelines.

Reply:
> It is valid: https://www.w3.org/TR/xhtml1/#h-4.6
> 
> Where do you get that it isn't?

Yes, _with_ slash is compliant for XHTML; _without_ slash isn't. But as 3D-I says, XHTML is obsolete anyway these days, and compliance it was never necessary even to run in old browsers. Modern HTML5 prefers _without_ slashes (but as before, using the with-slash style will still run fine in any modern browser).

HTML is very forgiving... until you forget to properly escape some user input, that is 😉